modules/ocs2/src/main/scala/gem/ocs2/Legacy.scala

```diff
@@ -139,7 +139,16 @@ object Legacy {
     }
 
     object Gnirs {
-      val WellDepth = Key("wellDepth")(Parsers.Gnirs.wellDepth)
+      import Parsers.Gnirs._
+      val Camera         = Key("camera"           )(camera)
+      val Decker         = Key("decker"           )(decker)
+      val Disperser      = Key("disperser"        )(disperser)
+      val DisperserOrder = Key("centralWavelength")(disperserOrder)
```

Comment:
There's an issue here, I think.  The central wavelength can be anything that the user types in.  It seems that you can go back from central wavelength to the corresponding disperser order using a function that tries to match them (see `GNIRSParams.Order.getOrder`) and failing that, picks a default value.

 Looking at the sequence XML, it seems that we don't even write the disperser order into the sequence.  We do write the wavelength though.  I think we need to update the GNIRS config to explicitly store the wavelength.  Since you cannot explicitly set the disperser order, we should probably remove that from the config and provide a function to compute it.

Reply:
Now it makes sense to me. In the OT UI the central wavelength the UI is a dropbox where you can either select a value or you can type directly the central wavelength, I didn't notice you could override the selectable values until now.